### PR TITLE
Support Markdown Metadata for components

### PR DESCRIPTION
**Pull request by eragon512**
Refer: #1361

### CODE REVIEW

File: src/scripts/logger.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /* eslint-disable no-console */
 
-const _ = require('lodash/fp');
+const _flow = require('lodash/fp/flow');
```
Comment (sapegin) [inline src/scripts/logger.js]: Any reason to change Lodash imports everywhere? Seems unrelated to the task but makes reviewing harder.

File: src/loaders/utils/processComponent.js [hunk 2]
```diff
@@ -28,14 +48,13 @@ module.exports = function processComponent(filepath, config) {
...
-	const componentMetadataPath = getComponentMetadataPath(filepath);
 	return {
 		filepath: componentPath,
 		slug: slugger.slug(componentName),
 		pathLine: config.getComponentPathLine(componentPath),
 		module: requireIt(filepath),
 		props: requireIt(`!!${propsLoader}!${filepath}`),
 		hasExamples: examplesFile && fs.existsSync(examplesFile),
-		metadata: fs.existsSync(componentMetadataPath) ? requireIt(componentMetadataPath) : {},
+		metadata: getComponentMetadata(filepath),
```
Comment (sapegin) [inline src/loaders/utils/processComponent.js]: It actually would make more sense to make it part of `props`. The current approach has multiple issues:

1. Metadata won't be hot reloaded on changes in Markdown file
2. We're reading a Markdown file one more time.
3. We're duplicating Markdown file name finding code.
4. We're doing some weird magic on client side, but we shouldn't send frontmatters to the browser at al.

File: src/scripts/logger.js (hunk 1 above)
Comment (eragon512) [reply src/scripts/logger.js]: will remove the commit for this :+1:

File: src/loaders/utils/processComponent.js (hunk 2 above)
Comment (eragon512) [reply src/loaders/utils/processComponent.js]: @sapegin so, should we keep everything (including the frontmatter) in `props` and remove the `metadata` prop above?

Comment (sapegin) [reply src/loaders/utils/processComponent.js]: Yup! `props` is quite stupid name, in some places we have `props.props.props` 🤦‍♂️ but I wound't date to change it before we migrate all the code to TypeScript ;-)

> Also, does the examplesFile variable have the markdown filepath? Didn't notice that, that's why I duplicated the Markdown file naming code 😅

Yeah, and we have it in props loader too:

https://github.com/styleguidist/react-styleguidist/blob/0ef7ff38a550b7b049bd0b82d87bd2ec97af3fce/src/loaders/props-loader.js#L69-L71

**Pull request closed without merging**